ci/do_ci.sh

```diff
@@ -218,11 +218,20 @@ elif [[ "$CI_TARGET" == "bazel.sizeopt" ]]; then
   bazel_binary_build sizeopt
   exit 0
 elif [[ "$CI_TARGET" == "bazel.gcc" ]]; then
-  BAZEL_BUILD_OPTIONS+=("--test_env=HEAPCHECK=")
+  # Temporariliy exclude some extensions from the envoy binary to address build failures
+  # due to long command line. Tests will still run.
+  BAZEL_BUILD_OPTIONS+=(
+    "--test_env=HEAPCHECK="
+    "--//source/extensions/filters/network/rocketmq_proxy:enabled=False"
+    "--//source/extensions/filters/http/admission_control:enabled=False"
+    "--//source/extensions/filters/http/dynamo:enabled=False"
+    "--//source/extensions/filters/http/header_to_metadata:enabled=False"
+    "--//source/extensions/filters/http/on_demand:enabled=False")
   setup_gcc_toolchain
 
-  echo "Testing ${TEST_TARGETS[*]}"
-  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" -c fastbuild "${TEST_TARGETS[@]}"
+  # Disable //test/config_test:example_configs_test so it does not fail because of excluded extensions above
+  echo "Testing ${TEST_TARGETS[*]} -//test/config_test:example_configs_test"
+  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" -c fastbuild -- "${TEST_TARGETS[@]}" -//test/config_test:example_configs_test
```

Comment:
I think fastbuild is hit the hardest by command line limits because of the longer bazel output paths of "fastbuild" while "opt" and "dbg" are considerably shorter: 6 characters shorter for each of the thousands of include and library paths.

Changing the gcc CI to use opt or dbg may serve as a decent work around until we find a better alternative.